### PR TITLE
Align List/Detail pages with PF

### PR DESCRIPTION
**Pull request by rawagner**
Improve Navigation active prop detection

Detail pages are more aligned with PF
Actions are on top right corner
Tabs are using react-router so you have exact links for specific tabs
details

List page
list

I've also reworked how Navigation is highlighted, the resulting style aligns with PF exactly, and it also works for Tabs (and possibly other components too)

### CODE REVIEW

File: src/app/components/NavItem/NavItem.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+type ActiveChildrenProps = {
+  isActive: boolean;
+  children: React.ReactNode;
+  setIsActive: (isActive: boolean) => void;
+};
+
+const ActiveChildren: React.FC<ActiveChildrenProps> = ({ isActive, children, setIsActive }) => {
+  React.useEffect(() => {
+    setIsActive(isActive);
```
Comment (celdrake) [inline src/app/components/NavItem/NavItem.tsx:14]: There's a lint warning seen in Github itself.
Can the dependency list be updated to include `setIsActive` too?

**Review by celdrake — APPROVED** [review]
Really great work! I love the encapsulation and the improved tab navigation!

Just a minor issue I noticed, is that the buttons of the empty state for Fleet / Repository have a surrounding white box.

We can fix this later if you prefer.

File: src/app/components/NavItem/NavItem.tsx (hunk 1 above)
Comment (rawagner) [reply src/app/components/NavItem/NavItem.tsx:14]: fixed

**Comment by rawagner** [comment]
I've fixed the white background in latest patch too.

**Pull request merged**